### PR TITLE
Fix: Result and Context arguments for component tests

### PR DESCRIPTION
**Pull request by sriteja777**
## Proposed changes
- The args context and result are getting converted to just object containing file due to [this](https://github.com/webdriverio/webdriverio/blob/0ac282e9849cf46f1cf9b82256fbbfa608da9754/packages/wdio-browser-runner/src/browser/frameworks/mocha.ts#L207). Added a check for type and title so that it adds file only in case of test and not for other args.
- The error in the arg is in instance of class Error but we need to convert it to a serializable object else it gets ignored during stringification. Added check for and error and converting to object (taken from [here](https://github.com/webdriverio/webdriverio/blob/8196c4ccb4f66e217e161804ef549f18b60dccb3/packages/wdio-mocha-framework/src/common.ts#L38))
- Check log line to debug

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Polish (an improvement for an existing feature)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (improvements to the project's docs)
- [ ] Specification changes (updates to WebDriver command specifications)
- [ ] Internal updates (everything related to internal scripts, governance documentation and CI files)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Backport Request

[//]: # (The current `main` branch is the development branch for WebdriverIO v9. If your change should be released to the current major version of WebdriverIO (v8), please raise another PR with the same changes against the `v8` branch.)

- [ ] This change is solely for `v9` and doesn't need to be back-ported
- [ ] Back-ported PR at `#XXXXX`

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-browser-runner/src/browser/frameworks/mocha.ts [hunk 1]
```diff
@@ -204,10 +204,33 @@ export class MochaFramework extends HTMLElement {
...
+                            matcherResult: arg.error.matcherResult,
+                            expected: arg.error.expected,
+                            actual: arg.error.actual
+                        }
+                        return {
+                            ...arg,
+                            error: errorObject
+                        }
+                    }
+                }
```
Comment (christian-bromann) [inline packages/wdio-browser-runner/src/browser/frameworks/mocha.ts:233]:
```suggestion
                if (typeof arg === 'object' && 'error' in arg && arg.error instanceof Error) {
                    const errorObject = {
                        // Pull all enumerable properties, supporting properties on custom Errors
                        ...arg.error,
                        // Explicitly pull Error's non-enumerable properties
                        message: arg.error.message,
                        name: arg.error.name,
                        stack: arg.error.stack,
                        type: arg.error.type || arg.error.name,
                        matcherResult: arg.error.matcherResult,
                        expected: arg.error.expected,
                        actual: arg.error.actual
                    }
                    return {
                        ...arg,
                        error: errorObject
                    }
                }
```

File: packages/wdio-browserstack-service/src/insights-handler.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ import {
     getUniqueIdentifier,
     getUniqueIdentifierForCucumber,
     isBrowserstackSession,
-    isScreenshotCommand,
+    isScreenshotCommand, isUndefined,
     o11yClassErrorHandler,
```
Comment (christian-bromann) [inline packages/wdio-browserstack-service/src/insights-handler.ts:20]:
```suggestion
    isScreenshotCommand,
    isUndefined,
    o11yClassErrorHandler,
```

**Review by christian-bromann — COMMENTED** [review]
Can we have a PR for the `main` branch please?

**Comment by sriteja777** [comment]
@christian-bromann Sorry for the delay. Have created PRs for main and v7.
- https://github.com/webdriverio/webdriverio/pull/12847
- https://github.com/webdriverio/webdriverio/pull/12848

As we don't support component testing in V7, have just ported other changes.

**Review by christian-bromann — APPROVED** [review]
Thanks a lot 👍

**Pull request merged**